android/app/src/main/java/io/github/parkwithease/parkeasy/ui/login/LoginViewModel.kt

```diff
@@ -30,40 +37,144 @@ constructor(
 
     val loggedIn = authRepo.statusFlow
 
+    var formState by mutableStateOf(LoginFormState())
+        private set
+
     private val _formEnabled = MutableStateFlow(true)
     val formEnabled = _formEnabled.asStateFlow()
 
-    fun onLoginPress(email: String, password: String) {
+    fun onLoginPress() {
         viewModelScope.launch {
             _formEnabled.value = false
-            val result: Boolean = userRepo.login(LoginCredentials(email, password))
+            userRepo
+                .login(LoginCredentials(formState.email.value, formState.password.value))
+                .onSuccess { showSnackbar("Logged in successfully", null) }
+                .recoverRequestErrors("Login failed")
             _formEnabled.value = true
-            showSnackbar(if (result) "Logged in successfully" else "Error logging in", null)
         }
     }
 
-    fun onRegisterPress(name: String, email: String, password: String, confirmPassword: String) {
-        if (password == confirmPassword) {
-            viewModelScope.launch {
-                _formEnabled.value = false
-                val result: Boolean =
-                    userRepo.register(RegistrationCredentials(name, email, password))
-                _formEnabled.value = true
-                showSnackbar(if (result) "Registered successfully" else "Error registering", null)
-            }
-        } else {
-            viewModelScope.launch { showSnackbar("Passwords don't match", null) }
+    fun onRegisterPress() {
+        if (formState.password.value != formState.confirmPassword.value) return
+
+        viewModelScope.launch {
+            _formEnabled.value = false
+            userRepo
+                .register(
+                    RegistrationCredentials(
+                        formState.name.value,
+                        formState.email.value,
+                        formState.password.value,
+                    )
+                )
+                .onSuccess { showSnackbar("Registered successfully", null) }
+                .recoverRequestErrors("Error registering")
+            _formEnabled.value = true
         }
     }
 
-    fun onRequestResetPress(email: String) {
+    fun onRequestResetPress() {
         viewModelScope.launch {
-            val result = userRepo.requestReset(ResetCredentials(email))
-            showSnackbar(
-                if (result) "Reset email sent\nJk... we're working on it"
-                else "Error resetting password",
-                null,
-            )
+            _formEnabled.value = false
+            userRepo
+                .requestReset(ResetCredentials(formState.email.value))
+                .onSuccess { showSnackbar("Reset email sent\nJk... we're working on it", null) }
+                .recoverRequestErrors("Error resetting password")
+            _formEnabled.value = true
```

Comment:
The login form is disabled for the duration of the snackbar being shown instead of the API request, this is most noticeable on the forgot password screen

Reply:
![image](https://github.com/user-attachments/assets/0e5cfd8e-b923-4655-8ee7-39288a042b07)
![image](https://github.com/user-attachments/assets/bb7af0ad-0844-4eba-80d0-21302c3cb3e9)